Add config to disable logging row data from processing log + config docs (#2419)
confluentinc/ksql: docs/installation/server-config/config-reference.rst

    # Bind only to localhost.
    listeners=http://localhost:8088

KSQL Processing Log Settings
----------------------------

These configurations control the behavior of the :ref:`KSQL processing log <ksql_processing_log>`.

.. _ksql-processing-log-topic-auto-create:

-------------------------------------

> ksql.processing.log.topic.auto.create

Reviewer: we should be generating such docs _from_ the config... :(

(Though don't get me wrong - you adding the docs is great!)

Author: I looked at doing this at first and then punted on it. ConfigDef is set up to do this, but we need to do a few things:
- Integrate it into the sphinx build stuff, which is a fair bit of work - I think we'll need to add a build target that can generate the docs, then update the docs repo to use it for ksql and copy the artifact over to the ksql docs before running sphinx-build
- The format of the output is different from the rest of the config docs, so we'd want to do this for all the config docs. But we'd want to be sure to configure internal configs correctly in our code.

I think it makes sense to at least have docs first.

cc @JimGalasyn

Reviewer: > I think it makes sense to at least have docs first.

Absolutely. Just raising this as its not something that's getting any attention. It's always a 'would it be nice if...'. 

Added https://github.com/confluentinc/ksql/issues/2423

this property. If automatic processing log stream creation is enabled, KSQL uses this topic to back the
stream. By default, this property has the value ``<service id>ksql_processing_log``, where ``<service id>``
is the value of the :ref:`ksql-service-id` property.

.. _ksql-processing-log-topic-partitions:

------------------------------------
ksql.processing.log.topic.partitions
------------------------------------

If automatic processing log topic creation is enabled, KSQL creates the topic with number of partitions set
to the value of this property. By default, this property has the value ``1``.

.. _ksql-processing-log-replication-factor:

--------------------------------------------
ksql.processing.log.topic.replication.factor
--------------------------------------------

If automatic processing log topic creation is enabled, KSQL creates the topic with  number of replicas set
to the value of this property. By default, this property has the value ``1``.

.. _ksql-processing-log-stream-auto-create:

--------------------------------------
ksql.processing.log.stream.auto.create
--------------------------------------

> Toggles automatic processing log stream creation. If set to true, and running interactive mode on a new cluster,

Reviewer 2: ```suggestion
Toggles automatic processing log stream creation. If set to true, and KSQL is running in interactive mode on a new cluster,
```

> then KSQL will automatically create a processing log stream when it starts up. The name for the stream is the

Reviewer 2: ```suggestion
KSQL automatically creates a processing log stream when it starts up. The name for the stream is the
```

> value of the :ref:`ksql-processing-log-stream-name` property. The stream will be created over the topic set in

Reviewer 2: ```suggestion
value of the :ref:`ksql-processing-log-stream-name` property. The stream is created over the topic set in
```

------------------------------

If automatic processing log topic creation is enabled, KSQL sets the name of the topic to the value of
this property. If automatic processing log stream creation is enabled, KSQL uses this topic to back the
stream. By default, this property has the value ``<service id>ksql_processing_log``, where ``<service id>``
is the value of the :ref:`ksql-service-id` property.

.. _ksql-processing-log-topic-partitions:

------------------------------------
ksql.processing.log.topic.partitions
------------------------------------

If automatic processing log topic creation is enabled, KSQL creates the topic with number of partitions set
to the value of this property. By default, this property has the value ``1``.

.. _ksql-processing-log-replication-factor:

--------------------------------------------
ksql.processing.log.topic.replication.factor
--------------------------------------------

If automatic processing log topic creation is enabled, KSQL creates the topic with  number of replicas set
to the value of this property. By default, this property has the value ``1``.

.. _ksql-processing-log-stream-auto-create:

--------------------------------------

> ksql.processing.log.stream.auto.create

Author: The user could already have a topic around they want to use. I think right now we log an error. I'll add some documentation that points out that the topic needs to be created one way or another.
